acceptance/acceptance_test.go

```diff
@@ -229,1905 +99,1417 @@ func testWithoutSpecificBuilderRequirement(
-			t.Helper()
-			packageName := registryConfig.RepoName("test/package-" + h.RandString(10))
-			output, err := h.RunE(subjectPack("package-buildpack", packageName, "-p", absConfigPath, "--publish"))
-			h.AssertNil(t, err)
-			h.AssertContains(t, output, fmt.Sprintf("Successfully published package '%s'", packageName))
-			return packageName
-		}
-
-		assertImageExistsLocally := func(name string) {
-			t.Helper()
-			_, _, err := dockerCli.ImageInspectWithRaw(context.Background(), name)
-			h.AssertNil(t, err)
-
-		}
-
-		generateAggregatePackageToml := func(buildpackURI, nestedPackageName string) string {
-			t.Helper()
-			packageTomlData := fillTemplate(t,
-				filepath.Join(packFixturesDir, "package_aggregate.toml"),
-				map[string]interface{}{
-					"BuildpackURI": buildpackURI,
-					"PackageName":  nestedPackageName,
-				},
-			)
-			packageTomlFile, err := ioutil.TempFile(tmpDir, "package_aggregate-*.toml")
-			h.AssertNil(t, err)
-			_, err = io.WriteString(packageTomlFile, packageTomlData)
-			h.AssertNil(t, err)
-			h.AssertNil(t, packageTomlFile.Close())
-
-			return packageTomlFile.Name()
-		}
-
 		when("no --format is provided", func() {
 			it("creates the package as image", func() {
 				packageName := "test/package-" + h.RandString(10)
-				output, err := h.RunE(subjectPack("package-buildpack", packageName, "-p", filepath.Join(tmpDir, "package.toml")))
-				h.AssertNil(t, err)
-				h.AssertContains(t, output, fmt.Sprintf("Successfully created package '%s'", packageName))
-				defer h.DockerRmi(dockerCli, packageName)
 
-				assertImageExistsLocally(packageName)
+				output := pack.SuccessfulRunWithOutput(
+					"package-buildpack", packageName,
+					"-p", simplePackageConfigPath,
+				)
+
+				assert.Contains(output, fmt.Sprintf("Successfully created package '%s'", packageName))
+				defer containerManager.RemoveImages(packageName)
+
+				assert.ImageExistsLocally(packageName)
 			})
-		})
 
-		when("--format image", func() {
-			it("creates the package", func() {
+			it("creates the package as a local image", func() {
 				t.Log("package w/ only buildpacks")
-				nestedPackageName := packageBuildpackLocally(filepath.Join(tmpDir, "package.toml"))
-				defer h.DockerRmi(dockerCli, nestedPackageName)
-				assertImageExistsLocally(nestedPackageName)
-
-				t.Log("package w/ buildpacks and packages")
-				aggregatePackageToml := generateAggregatePackageToml("simple-layers-parent-buildpack.tgz", nestedPackageName)
-				packageName := packageBuildpackLocally(aggregatePackageToml)
-				defer h.DockerRmi(dockerCli, packageName)
-				assertImageExistsLocally(packageName)
+				nestedPackageName := newPackageName()
+				assert.PackageCreated(
+					nestedPackageName,
+					pack.PackageBuildpack(nestedPackageName, simplePackageConfigPath),
+				)
+				defer containerManager.RemoveImages(nestedPackageName)
+
+				assert.ImageExistsLocally(nestedPackageName)
+
+				aggregatePackageConfigFile := pack.AggregatePackageFixture(
```

Comment:
At first glance it's a little unclear to me what `pack.AggregatePackageFixture` is doing... I had to click around a bit and am still not 100% sure. It's a little strange to me that `PackExecutor` is doing test-type things (such as filling in templates) and "real" things (such as calling `pack package-buildpack`). I wonder if separating those concerns could make things a bit easier to follow.

Reply:
I would like to figure out how we can keep the concepts of the source of fixtures tightly coupled to the implementation of pack we're using. If we separate these, I can see a situation where it becomes confusing which fixture source we should be using for which version of pack.

There is, however, clearly a mixture of concerns here so there's some more thought required to balance tightly coupling execution and fixture source with separation of command execution.